metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/ExternalFunctions.java

```diff
@@ -0,0 +1,292 @@
+                assert pb.redirectOutput().file() == log;
+                assert p.getInputStream().read() == -1;
+
+                return exitCode;
+            }
+            catch (NullPointerException e) {
+                System.err.println("Error: " + e.toString());
+                return null;
+            }
+            catch (IndexOutOfBoundsException e) {
+                System.err.println("Error: " + e.toString());
+                return null;
+            }
+            catch (SecurityException e) {
+                System.err.println("Error: " + e.toString());
+                return null;
+            }
+            catch (IOException e) {
+                System.err.println("Error: " + e.toString());
+                return null;
+            } catch (InterruptedException e) {
+                System.err.println("Error: " + e.toString());
+                return null;
+            }
+        }
+	
+	    @Override
+        public void initialize(Context context) {
+
+        }
+	
+	    @Override
+	    public boolean isInitialized() {
+            return true;
+	    }
+	}
+
+    /**
+     * Special thanks to alvin j. alexander at devdaily.com
+     */
+    class ThreadedStreamHandler extends Thread
+    {
+        InputStream inputStream;
+        String adminPassword;
+        OutputStream outputStream;
+        PrintWriter printWriter;
+        StringBuilder outputBuffer = new StringBuilder();
+        private boolean sudoIsRequested = false;
+
+        /**
+         * A simple constructor for when the sudo command is not necessary.
+         * This constructor will just run the command you provide, without
+         * running sudo before the command, and without expecting a password.
+         *
+         * @param inputStream
+         */
+        ThreadedStreamHandler(InputStream inputStream)
+        {
+            this.inputStream = inputStream;
+        }
+
+        /**
+         * Use this constructor when you want to invoke the 'sudo' command.
+         * The outputStream must not be null. If it is, you'll regret it. :)
+         *
+         * TODO this currently hangs if the admin password given for the sudo command is wrong.
+         *
+         * @param inputStream
+         * @param outputStream
+         * @param adminPassword
+         */
+        ThreadedStreamHandler(InputStream inputStream, OutputStream outputStream, String adminPassword)
+        {
+            this.inputStream = inputStream;
+            this.outputStream = outputStream;
+            this.printWriter = new PrintWriter(outputStream);
+            this.adminPassword = adminPassword;
+            this.sudoIsRequested = true;
+        }
+
+        public void run()
+        {
+            // on mac os x 10.5.x, when i run a 'sudo' command, i need to write
+            // the admin password out immediately; that's why this code is
+            // here.
+            if (sudoIsRequested)
+            {
+                //doSleep(500);
```

Comment:
This is getting into some 'Expect' type stuff, there be dragons here.  What if this hangs forever?